code/datums/gamemode/dynamic/dynamic.dm

```diff
@@ -0,0 +1,87 @@
+/datum/gamemode/dynamic
+	name = "Dynamic Mode"
+	var/threat_level = 0//rolled at the beginning of the round.
+	var/threat = 0//set at the beginning of the round. Spent by the mode to "purchase" rules.
+	var/list/roundstart_rules = list()
+	var/list/second_rule_req = list(100,100,100,80,60,40,20,0,0,0)//requirements for extra round start rules
+	var/list/third_rule_req = list(100,100,100,100,100,70,50,30,10,0)
+	var/roundstart_pop_ready = 0
+	var/list/candidates = list()
+	var/list/current_rules = list()
+	var/list/executed_rules = list()
+
+/datum/gamemode/dynamic/can_start()
+	threat_level = rand(1,100)*0.6 + rand(1,100)*0.4//https://docs.google.com/spreadsheets/d/1QLN_OBHqeL4cm9zTLEtxlnaJHHUu0IUPzPbsI-DFFmc/edit#gid=499381388
+	threat = threat_level
+	message_admins("Dynamic Mode initialized with a Threat Level of... <font size='8'>[threat_level]</font>!")
+	return 1
+
+/datum/gamemode/dynamic/Setup()
+	for (var/rule in subtypesof(/datum/dynamic_ruleset/roundstart))
+		roundstart_rules += new rule()
+	for(var/mob/new_player/player in player_list)
+		if(player.ready && player.mind)
+			roundstart_pop_ready++
+			candidates.Add(player)
+	message_admins("Listing [roundstart_rules.len] round start rulesets, and [candidates.len] players ready.")
+	if (candidates.len <= 0)
+		message_admins("Not a single player readied-up. The round will begin without any roles assigned.")
+		return 1
+	if (roundstart_rules.len <= 0)
+		message_admins("There are no roundstart rules within the code, what the fuck? The round will begin without any roles assigned.")
```

Comment:
Keep it semi-professional. No whadafuk in the code comments

Reply:
This is not a policy we have.

Reply:
It's an admin-only message, and it's not a professional game.